source/blink.cpp

```diff
@@ -133,30 +153,48 @@ void blink::application::run()
 		print("  Started process with PID " + std::to_string(pi.dwProcessId));
 	}
 
-	print("Starting file system watcher for '" + _source_dir.string() + "' ...");
+	size_t dir_index = 0;
+	std::vector<scoped_handle> dir_handles;
+	std::vector<scoped_handle> event_handles;
+	std::vector<notification_info> notification_infos;
+	for (auto it = _source_dirs.begin(); it != _source_dirs.end(); ++it) {
+		auto source_dir = *it;
+		print("Starting file system watcher for '" + source_dir.string() + "' ...");
 
-	// Open handle to the common source code directory
-	const scoped_handle watcher_handle = CreateFileW(_source_dir.native().c_str(), FILE_LIST_DIRECTORY, FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, nullptr, OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS, nullptr);
+		dir_handles.push_back(INVALID_HANDLE_VALUE);
+		event_handles.push_back(INVALID_HANDLE_VALUE);
+		notification_info info;
+		notification_infos.push_back(info);
```

Comment:
These are better off with `emplace_back`:
```c++
dir_handles.emplace_back();
event_handles.emplace_back();
notification_infos.emplace_back();
```

Reply:
Thanks. Using .emplace_back() calls now.

Reply:
Nono, `emplace_back` calls the constructor of the item and passes in whatever arguments you pass to `emplace_back`. So you don't need to create any temporaries, but rather can just default-construct them in-place:
Can just write
```
dir_handles.emplace_back();
event_handles.emplace_back();
notification_infos.emplace_back();
```
instead of
```
dir_handles.emplace_back(INVALID_HANDLE_VALUE);
event_handles.emplace_back(INVALID_HANDLE_VALUE);
notification_info info;
notification_infos.emplace_back(info);
```
Otherwise there would be no point in replacing `push_back` with `emplace_back`.

Reply:
Thanks. Fixed.